nucleus/common/internal-api/src/main/java/fish/payara/monitoring/collect/MonitoringDataCollector.java

```diff
@@ -88,14 +109,26 @@ default MonitoringDataCollector prefix(CharSequence prefix) {
         MonitoringDataCollector self = this;
         return new MonitoringDataCollector() {
 
+            StringBuilder prefixed = new StringBuilder(prefix);
+
             @Override
             public MonitoringDataCollector tag(CharSequence name, CharSequence value) {
-                return self.tag(name, value);
+                self.tag(name, value);
+                return this;
+            }
+
+            @Override
+            public MonitoringDataCollector collect(CharSequence metric, long value) {
+                prefixed.setLength(prefix.length());
+                self.collect(prefixed.append(metric), value);
+                return this;
             }
 
             @Override
-            public MonitoringDataCollector collect(CharSequence key, long value) {
-                return self.collect(new StringBuilder(prefix).append(key), value);
+            public MonitoringDataCollector annotate(CharSequence metric, long value, String... attrs) {
+                prefixed.setLength(prefix.length());
+                self.annotate(prefixed.append(metric), value, attrs);
+                return this;
```

Comment:
NB. It was a bug to not return `this` for any of the methods of `MonitoringDataCollector` because when the instance is used with chaining it should be the wrapper that is called (so it does its prefix thing), not the instance returned by `self` (that would not do the prefix thing). In practice this bug never had an effect as the only usage would not use chaining but for future this now works properly.